models/netsuite2/netsuite2__transaction_details.sql

```diff
@@ -188,7 +200,10 @@ transaction_details as (
     on accounting_periods.accounting_period_id = transactions.accounting_period_id
 
   left join customers 
-    on customers.customer_id = coalesce(transaction_lines.entity_id, transactions.entity_id)
+    on case when lower(transactions.transaction_type) in ('custinvc', 'custcred')
+        then customers.customer_id = transactions.entity_id
```

Comment:
@jmongerlyra are there any cases where the transaction type is `'custinvc', 'custcred', 'vendbill', or 'vendcred'` but the transaction `entity_id` is null? 

essentially i am wondering if there is a need for a `coalesce(transactions.entity_id, transaction_lines.entity_id)` here

Reply:
@fivetran-jamie `transactions.entity_id` is on header record of the transaction, and the ID master table is contingent on the transaction type. For 'custinvc' & 'custcred' it's `customers.customer_id`. It's the vendor table for bills.

`coalesce(transactions.entity_id, transaction_lines.entity_id)` is needed because on journal entries (different trans type) a customer/vendor ID on `transaction_lines.entity_id`

Reply:
@fivetran-jamie This patch is slower than we would like, so I have rebased it out of my fork. The PR is back where it was originally. I'll submit a separate PR for issue [109](https://github.com/fivetran/dbt_netsuite/issues/109). No further changes will be made here.